### PR TITLE
Abstracting out vulkan resource reader class

### PR DESCRIPTION
**Pull request by panos-lunarg**

### CODE REVIEW

**Comment by panos-lunarg** [comment]
I have a disagreement between my `clang-format` and the one used by the checker although I believe I am using the correct version (9)

**Comment by bradgrantham-lunarg** [comment]
> I have a disagreement between my `clang-format` and the one used by the checker although I believe I am using the correct version (9)

You can cut-and-paste the diff out of the CI job log and apply it as a patch.  Let me know if I can help with detail on how to do that.

File: framework/graphics/vulkan_resources_util.cpp [hunk 1]
```diff
@@ -0,0 +1,1352 @@
...
+        create_info.arrayLayers           = all_layers_per_level ? array_layers : 1;
+        create_info.samples               = VK_SAMPLE_COUNT_1_BIT;
+        create_info.tiling                = tiling;
+        create_info.usage                 = VK_IMAGE_USAGE_TRANSFER_SRC_BIT;
+        create_info.sharingMode           = VK_SHARING_MODE_EXCLUSIVE;
+        create_info.queueFamilyIndexCount = 0;
+        create_info.pQueueFamilyIndices   = nullptr;
+        create_info.initialLayout         = VK_IMAGE_LAYOUT_UNDEFINED;
+
+        for (uint32_t m = 0; m < mip_levels; ++m)
```
Comment (davidlunarg) [inline framework/graphics/vulkan_resources_util.cpp:327]: Should the value of mip_levels be validated before it is used?

**Review by davidlunarg — COMMENTED** [review]
This is a big change, in code I have not previously visited, but I attempted to review this anyway.

I couldn't get the big picture of what the changes were without diving into code details. Function headers and a description of parameters might help. (But I realize most of gfxrecon is written with minimal function headers and param descriptions.)

After some effort, I did figure out the big picture. That looks good. I think.

I would feel better if someone else also reviewed this code before approving it.

**Comment by panos-lunarg** [comment]
>I couldn't get the big picture of what the changes were without diving into code details. Function headers and a description of parameters might help. (But I realize most of gfxrecon is written with minimal function headers and param descriptions.)

Good point. I'll add some description in functions and their parameters

**Review by bradgrantham-lunarg — APPROVED** [review]

**Comment by panos-lunarg** [comment]
oops forgot to squash and merge...

**Pull request merged**